### ISSUE TITLE
TFLint fails with not enough function arguments for nested modules with provider functions

### ISSUE THREAD

**Issue by MrKich** [terraform-linters/tflint#2339]
### Summary

Found an error while using deepmerge function from provider in nested modules:
```
  input1 = provider::deepmerge::mergo({}, {}, {}, {})
  input2 = provider::deepmerge::mergo({}, {})
```

`Function "provider::deepmerge::mergo" expects 4 argument(s). Missing value for "".:`

If you make the number of arguments the same, then error is gone:
```
  input1 = provider::deepmerge::mergo({}, {}, {}, {}, {})
  input2 = provider::deepmerge::mergo({}, {}, {}, {}, {})
```

terraform plan/apply works correctly.

### Command

tflint

### Terraform Configuration

```terraform
# ./main.tf
terraform {
  required_version = ">=1.9"
}

module "mod-parent" {
  source = "./mod-parent"

  map = {
    myobject = {}
  } 
}



# ./mod-parent/main.tf
terraform {
  required_version = ">=1.9"
  required_providers {
    deepmerge = {
      source  = "isometry/deepmerge"
      version = "~> 1.0.2"
    }
  }
}

variable "map" {
  type = any
}

module "mod-child" {
  for_each = var.map
  source = "../mod-child"

  # Fails here. If you remove for_each, or make the number of arguments below the same, then the error is gone
  input1 = provider::deepmerge::mergo({}, {}, {}, {})
  input2 = provider::deepmerge::mergo({}, {})
}



# ./mod-child/main.tf
variable "input1" {
  type = any
}

variable "input2" {
  type = any
}
```

### TFLint Configuration

```HCL
*default
```

### Output

```Shell
14:16:33 config.go:178: [INFO] Load config: .tflint.hcl
14:16:33 config.go:186: [INFO] file not found
14:16:33 config.go:193: [INFO] Load config: /home/mrkich/.tflint.hcl
14:16:33 config.go:201: [INFO] file not found
14:16:33 config.go:204: [INFO] Use default config
14:16:33 config.go:464: [INFO] The "terraform" plugin block is not found. Enable the plugin "terraform" automatically
14:16:33 option.go:77: [DEBUG] CLI Options
14:16:33 option.go:78: [DEBUG]   CallModuleType: local
14:16:33 option.go:79: [DEBUG]   Force: false
14:16:33 option.go:80: [DEBUG]   Format:
14:16:33 option.go:81: [DEBUG]   Varfiles:
14:16:33 option.go:82: [DEBUG]   Variables:
14:16:33 option.go:83: [DEBUG]   EnableRules:
14:16:33 option.go:84: [DEBUG]   DisableRules:
14:16:33 option.go:85: [DEBUG]   Only:
14:16:33 option.go:86: [DEBUG]   EnablePlugins:
14:16:33 option.go:87: [DEBUG]   IgnoreModules:
14:16:33 loader.go:39: [INFO] Initialize new loader
14:16:33 module_mgr.go:63: [INFO] Module manifest file found. Initializing...
14:16:33 loader.go:81: [INFO] Building the root module while calling local child modules...
14:16:33 loader.go:108: [DEBUG] Trying to load the local module: name=mod-parent dir=mod-parent
14:16:33 loader.go:108: [DEBUG] Trying to load the local module: name=mod-child dir=mod-child
14:16:33 runner.go:47: [INFO] Initialize new runner for root
14:16:33 runner.go:47: [INFO] Initialize new runner for module.mod-parent
Failed to prepare rule checking; mod-parent/main.tf:20,45-46: Not enough function arguments; Function "provider::deepmerge::mergo" expects 4 argument(s). Missing value for "".:

Error: Not enough function arguments

  on mod-parent/main.tf line 20, in module "mod-child":
  20:   input2 = provider::deepmerge::mergo({}, {})

Function "provider::deepmerge::mergo" expects 4 argument(s). Missing value for "".
```

### TFLint Version

0.58.0

### Terraform Version

v1.9.5

### Operating System

- [x] Linux
- [ ] macOS
- [ ] Windows
